processor/geoipprocessor/geoip_processor.go

```diff
@@ -5,26 +5,117 @@ package geoipprocessor // import "github.com/open-telemetry/opentelemetry-collec
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"net"
 
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/geoipprocessor/internal/provider"
 )
 
-type geoIPProcessor struct{}
+var errIPNotFound = errors.New("no IP address found in the resource attributes")
+
+// newGeoIPProcessor creates a new instance of geoIPProcessor with the specified fields.
+type geoIPProcessor struct {
+	providers []provider.GeoIPProvider
+	fields    []string
+}
+
+func newGeoIPProcessor(fields []string) *geoIPProcessor {
```

Comment:
Why not pass the whole config into the processor? It seems more natural to me to encapsulate the configuration value in the config struct, and it will probably be needed later anyway.

Reply:
I [replaced](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33319/commits/33ab2f5deb72d79fb21e985e200aeb93c3be67c0) this configuration with a resource_attributes array, a default value with the `source.address` key is provided by the processor's factory. What do you think?